hyperspy/roi.py

```diff
@@ -322,11 +322,13 @@ def interactive(self, signal, navigation_signal="same", out=None,
             If not None, it will use 'out' as the output instead of returning
             a new Signal.
         """
-        if navigation_signal == "same":
+        if isinstance(navigation_signal, basestring) and navigation_signal == "same":
             navigation_signal = signal
         if navigation_signal is not None:
             if navigation_signal not in self.signal_map:
                 self.add_widget(navigation_signal)
+        if self.update not in signal.events.data_changed.connected:
+            signal.events.data_changed.connect(self.update, [])
```

Comment:
> when moving it, the line (eventually, if moved too far) collapses into a point that is still reachable by the user.

For this behavior, you would only need to connect to `signal.axes_manager.events.any_axis_changed`, as it is the offset change that is important, not the `data_changed` event. When connecting to `data_changed`, we will have a lot of redundant calls to `update()`. Either way, the ROI should not be limited by the signal bounds (as they are signal agnostic), only widgets or `__call__()` should care about bounds for passed signals.

Regarding the example (rectangle -> line), it doesn't really make sense out of the box. I guess the point is to illustrate that a fixed slicing of a moving rect won't work? In that case I would much prefer an example that shows how to set up a working version that does that. Coding blind (i.e. not tested):
```python
import scipy.misc
im = hs.signals.Image(scipy.misc.ascent())
roi = hs.roi.RectangularROI(left=275, right=360, top=300, bottom=380)
roi2 = hs.roi.Line2DROI(300, 345, 324, 353)
im.plot()
imr = roi.interactive(im)
imr.plot()

def make_stationary(signal, axes=None, offset=None, out=None):
     axes = # Get axes normal way
     if offset is None:
          offset = (0,) * len(axes)
     if out:
        s = out
        s.data = signal.data  # Reference data directly since we're in control
     else:
        s = signal._deepcopy_with_new_data(signal.data)
     for i, ax in enumerate(axes):
          ax.offset = offset[i]   # Indexing might need adjustment to fit input order
     if out:
          out.events.data_changed.trigger(out)
     else:
          return s

imr_stationary = hs.interactive(make_stationary, None, imr.axes.events.any_axes_changed, signal=imr, axes=<value here>)
# TODO: Francisco removed way of not giving hs.interactive an event (None -> auto to data_changed), so remove that connection (can't think of the code from the top of my head),
# Directly wire imr.events.data_changed:
imr.events.data_changed.connect(imr_stationary.events.data_changed.trigger, [])  # Use care if signal keyword is needed
imr2 = roi2.interactive(imr_stationary)
imr2.plot()
```